test/Datadog.Trace.Tests/SpanTests.cs

```diff
@@ -91,51 +91,35 @@ public void Finish_EndTimeInThePast_DurationIs0()
             Assert.Equal(TimeSpan.Zero, span.Duration);
         }
 
-        [Theory]
-        [InlineData(3)]
-        [InlineData(10)]
-        [InlineData(100)]
-        public void Accurate_Duration(int minimumSleepMilliseconds)
+        [Fact]
+        public void Accurate_Duration()
         {
-            // TODO: refactor how we measure time so we can lower this threshold
+            // Check that span has the same precision as stopwatch
+
             const int iterations = 10;
-            const int maxDifference = 15;
-            TimeSpan totalStopwatchTime = TimeSpan.Zero;
-            TimeSpan totalSpanTime = TimeSpan.Zero;
-            Span span;
             var stopwatch = new Stopwatch();
 
-            // execute once to ensure JIT compilation
-            using (span = _tracer.StartSpan("Operation"))
+            for (int i = 0; i < iterations; i++)
             {
-                stopwatch.Restart();
-                Thread.Sleep(1);
-                var spanMilliseconds = span.Duration.TotalMilliseconds;
-                var elapsedMilliseconds = stopwatch.ElapsedMilliseconds;
-            }
+                Span span;
 
-            // execute multiple times and average the results
-            for (int x = 0; x < iterations; x++)
-            {
                 using (span = _tracer.StartSpan("Operation"))
                 {
                     stopwatch.Restart();
-                    Thread.Sleep(minimumSleepMilliseconds);
-                    totalStopwatchTime += stopwatch.Elapsed;
+                    stopwatch.Stop();
                 }
 
-                totalSpanTime += span.Duration;
-            }
+                span.Duration.Should().BeGreaterOrEqualTo(stopwatch.Elapsed);
 
-            var avgStopwatch = totalStopwatchTime.TotalMilliseconds / iterations;
-            var avgSpan = totalSpanTime.TotalMilliseconds / iterations;
-            var stopwatchSpanDiff = Math.Abs(avgSpan - avgStopwatch);
-            // The difference should be less than the threshold
-            _output.WriteLine($"Average elapsed time: {avgStopwatch:0.0} ms");
-            _output.WriteLine($"Average span time: {avgSpan:0.0} ms");
-            Assert.True(
-                stopwatchSpanDiff < maxDifference,
-                $"Span duration difference ({stopwatchSpanDiff}ms) outside of allowed threshold {maxDifference}ms. Expected average: {avgStopwatch}ms, actual average: {avgSpan}ms");
+                stopwatch.Restart();
+                using (span = _tracer.StartSpan("Operation"))
+                {
+                }
```

Comment:
Any concerns with the span duration here being as short as possible? Should we create fake work so there's at least _some_ duration?

Reply:
It seems unlikely to happen, but it's true that the test is useless if the work is too short to be measured with `Stopwatch`. Gonna add a Thread.Sleep just to be sure

Reply:
Done in https://github.com/DataDog/dd-trace-dotnet/pull/1360/commits/22f40459215c1c19c33e5469a4e92bc3c1d60820